### ISSUE TITLE
[member-naming] Avoid for 'constructor' of class

### ISSUE THREAD

**Issue by saneyuki** [typescript-eslint/typescript-eslint#359]
**Repro**

```JSON
{
  "rules": {
        "@typescript-eslint/member-naming": ["warn", {
            "private": "^_",
            "protected": "^_",
        }],
  }
}
```

```TS
class A {
   private constructor() {}
}

class B {
   protected constructor() {}
}
```

**Expected Result**

`@typescript-eslint/member-naming` should avoid the warn/error if the method is `constructor`.

**Actual Result**

```
warning  private property constructor should match /^_/  @typescript-eslint/member-naming
warning   property constructor should match /^_/  @typescript-eslint/member-naming
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.2` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.2` |
| `node`                             | `11.6.0` |
| `npm`                              | `X.Y.Z` |

**Comment by gavinbarron** [comment]
I'll take a look at this one
